spec/services/fixed_region_info_spec.rb

```diff
@@ -0,0 +1,125 @@
+require "rails_helper"
+
+ describe FixedRegionInfo do
+   it "calculates average, number of contributors and number of samples for sessions for given stream for last hour" do
+    user1 = create_user!(id: 1)
+    user2 = create_user!(id: 2)
+    session1 = create_session_with_streams_and_measurements!(id: 1, value: 1, user: user1, count: 60)
+    session2 = create_session_with_streams_and_measurements!(id: 2, value: 2, user: user2, count: 60)
+
+    data = {
+      session_ids: [session1.id, session2.id],
+      sensor_name: "AirBeam2-F",
+    }
+
+    region_info = FixedRegionInfo.new.call(data)
+
+    expect(region_info[:average]).to eq(1.5)
+    expect(region_info[:number_of_contributors]).to eq(2)
+    expect(region_info[:number_of_samples]).to eq(120)
+    expect(region_info[:number_of_instruments]).to eq(2)
+  end
+
+  it "doesn't take into account measurements older than one hour" do
+    user1 = create_user!(id: 1)
+    user2 = create_user!(id: 2)
+    session1 = create_session_with_streams_and_measurements!(id: 1, value: 1, user: user1, count: 60)
+    session2 = create_session!(id: 2, user: user2)
+    stream = create_stream!(session: session2)
+    create_measurements!(value: 2, stream: stream, count: 60)
+    create_old_measurements!(value: 20, stream: stream)
+
+    data = {
+      session_ids: [session1.id, session2.id],
+      sensor_name: "AirBeam2-F",
+    }
+
+    region_info = FixedRegionInfo.new.call(data)
+
+    expect(region_info[:average]).to eq(1.5)
+    expect(region_info[:number_of_contributors]).to eq(2)
+    expect(region_info[:number_of_samples]).to eq(120)
+    expect(region_info[:number_of_instruments]).to eq(2)
+  end
+end
+
+ private
+
+ def create_session_with_streams_and_measurements!(attributes)
+  session = create_session!(id: attributes.fetch(:id), user: attributes.fetch(:user))
+  stream = create_stream!(session: session, min_latitude: attributes.fetch(:min_latitude, 1.0), max_latitude: attributes.fetch(:max_latitude, 1.0), min_longitude: attributes.fetch(:min_longitude, 1.0), max_longitude: attributes.fetch(:max_longitude, 1.0))
+  create_measurements!(stream: stream, value: attributes.fetch(:value), count: attributes.fetch(:count))
+
+   session
+end
+
+ def create_session!(attributes)
```

Comment:
We should prolly be extracting those `create_*` to a common place

Reply:
I thought that keeping them in the same file gives us the advantage of seeing where everything comes from and what exactly is happening. 
If we move them somewhere else then why not just use the factory girl (or upgrade to factory bot)?

Reply:
I think it's explicit enough what happens: we explicitly pass the parameters that we need in the created object.

I'd personally advise against FactoryGirl. Also the authors of the gem say
> My general advice, though, is to avoid Factory Girl as much as is reasonably possible. Not because it’s bad or unreliable software (Factory Girl is very reliable; we’ve used it successfully since 2008), but because its inherent persistence mechanism is calling #save! on the object, which will always take longer than not persisting data.
https://thoughtbot.com/blog/speed-up-tests-by-selectively-avoiding-factory-girl

Reply:
I'm still don't see why using factory bot would be a bad thing? In this file, we are using `create!` everywhere, which executes `save`, using not using factory girl doesn't improve the tests speed?

It's probably a sign of a bad code structure, cause we are executing sql queries all over the code. If we ware using a repository in that code, we could inject the repo in this test and the just do `Object.new`. But that would be a bigger refactor.

Reply:
The problem with FactoryGirl is that it hides database queries (too) conveniently. In many projects I've seen situations where a single Factory.create! fired 5+ queries when actually only one was needed. That situation times many unit tests makes the entire suite incredibly slow (for no reasons).